### ISSUE TITLE
Rewrite the commit history to remove tooling trailers

### ISSUE THREAD

**Issue by Dayallenr** [Dayallenr/RAG#31]
## Parent

#26 — Present the project to a hiring audience without weakening its claims

## Problem Statement

The commit history is the one public surface the rest of #26 cannot reach.
Fourteen of twenty-seven commits carry a tooling co-authorship trailer. Five
commit bodies reference the working-notes file in prose, and one commit title
names it outright — so even after the file is untracked, the history names it
and the reference dead-ends.

A reviewer who runs `git log` after reading a polished README meets a different
story than the one the repository tells, and the mismatch is worse than either
surface alone.

The history is otherwise an asset. Twenty-seven commits with substantive messages
show incremental work over time, which is what a reviewer wants to see and what
a single squashed commit destroys.

## Solution

Rewrite the history once, at the end of #26, after all other work is committed.
Strip the trailers, repoint the prose references at `docs/engineering-notes.md`,
and keep all twenty-seven commits with their messages and content otherwise
intact. Review the rewritten log before anything leaves the machine, then
force-push on an explicit go-ahead.

## User Stories

1. As the author, I want the history free of tooling co-authorship trailers, so that
   the public record reflects how I present the work.
2. As the author, I want commit prose to reference documents that exist, so that
   the history is not the only place naming an untracked file.
3. As a reviewer running `git log`, I want incremental, well-described commits, so
   that I can see sustained engineering.
4. As a reviewer, I want the commit count and granularity preserved, so that the
   history is not a single opaque dump.
5. As a reviewer, I want commit messages to still explain why each change was
   made, so that the rewrite loses no engineering content.
6. As the author, I want to review the rewritten log before it is published, so
   that I approve the exact public record.
7. As the author, I want the rewrite to happen once, so that a single irreversible
   operation covers all of the epic's work.
8. As the author, I want new commits to stop adding trailers immediately, so that
   nothing accumulates between now and the rewrite.
9. As the author, I want the tree at `HEAD` to be byte-identical before and after,
   so that I can prove the rewrite changed messages only.

## Implementation Decisions

- Trailer suppression for new commits is configured at the start of #26, not at
  the end. The rewrite then covers everything in one pass.
- The rewrite edits commit messages only. All twenty-seven commits survive;
  squashing is rejected because an incremental history is evidence of sustained
  work and a one-commit repository is a warning sign in itself.
- Prose references to the working-notes file, in five bodies and in the one
  title, are repointed at `docs/engineering-notes.md`, so the history stays
  coherent with the repository.
- Author and committer identity, dates, and tree contents are unchanged.
- The rewritten log is presented for review in full before any push.
- The force-push is a hard gate: it does not happen without an explicit real-time
  go-ahead. It is the last action of the epic.
- Only branch `main` exists to rewrite, and the repository is solo-owned, so the
  usual cost of a force-push — collaborators diverging — is not present.

## Testing Decisions

No code seam. Verification is mechanical and should be done before the push:

- The tree at `HEAD` after the rewrite is identical to the tree before it, so the
  rewrite provably touched messages only.
- The commit count is unchanged at twenty-seven.
- A search across all commit messages for the trailer and for the old filename
  returns nothing.
- The working tree is clean and the full suite still passes at `HEAD`.

## Out of Scope

- Squashing, reordering, or dropping commits.
- Rewriting file contents anywhere in history, including purging the large
  tracked artifacts from the pack.
- Changing author identity or commit dates.

## Further Notes

This is the last ticket in #26 and must not be started until every other child is
merged, because any commit made afterwards would need a second rewrite.

Force-pushing rewritten history to a public repository is irreversible in
practice. The pre-push verification above exists so the operation is auditable
rather than trusted.

**Comment by Dayallenr** [comment]
## Rewrite prepared locally and verified — awaiting the force-push go-ahead (2026-08-18)

Done on the machine, nothing published. `main` is rewritten; `origin/main` is untouched
and the two have diverged 40/41.

**Recovery is a one-liner if this is rejected:** `git reset --hard backup/pre-rewrite-31`
(also tagged `pre-rewrite-31`, and `refs/original/refs/heads/main` from filter-branch).

### The counts in the problem statement had drifted

The ticket says "fourteen of twenty-seven commits carry a tooling co-authorship trailer".
As of today it was **13 of 40** (41 with the commit that landed alongside this).
Trailer text was uniform: `Co-Authored-By: Claude Opus 5 <noreply@anthropic.com>`, always
the last line, always preceded by a blank line.

### The prose-reference criterion was already satisfied

The ticket also says "five commit bodies reference the working-notes file in prose, and
one commit title names it outright". Neither holds now. **No commit title names it**, and
exactly one body mentions it — `Publish the debugging log as documentation` — where the
mention is accurate historical narrative that already names its replacement:

> All of it lived in a working-notes file addressed to a tool, which about thirteen
> docstrings cited by name. docs/engineering-notes.md carries it as a debugging log
> instead, and the docstrings point there.

That is not a dead-end reference, so it was left alone. Checked mechanically rather than
by eye: every `*.md` path named in any commit body was resolved against the working tree,
and **none is missing**.

### What the rewrite changed, and the proof it changed nothing else

`git filter-branch --msg-filter` stripping only trailer lines and any blank line left
behind. Verified by comparing before/after:

| property | result |
|---|---|
| commit count | 41 → 41 |
| tree, author, author date, committer, committer date, subject — all 41 | byte-identical |
| `HEAD^{tree}` | `11c17c7a` before and after |
| `git diff backup/pre-rewrite-31 HEAD` | empty |
| bodies differing beyond the stripped trailer | 0 of 41 |
| merge commits | 4 preserved |
| `Refs #N` trailers | preserved |
| remaining tooling trailers | 0 |

`ruff check` clean, 394 tests pass.

### Acceptance status

- [x] Trailers stripped from every commit
- [x] All commits kept, messages and content otherwise intact
- [x] Prose references resolve (verified: no missing file referenced anywhere in the log)
- [x] The rewritten log reviewed before anything leaves the machine
- [ ] Force-push on an explicit go-ahead — **gated, not done**

### One thing outside the stated solution, flagged not done

Five commits carry placeholder subjects: `update` (×2), `answers`, `docs`, `queries`.
They sit against user stories 3 and 5, which ask for incremental, well-described commits
that explain why each change was made. The ticket's Solution says to keep messages
"otherwise intact", so they were not touched. Rewording them would cost nothing extra
*now* — the rewrite is already staged — whereas doing it later means a second force push.
Author's call.

**Comment by Dayallenr** [comment]
## State check, 2026-08-18 — the rewrite was undone by a rebase, not pushed

Verified against the refs rather than taken from the working notes, which claimed
this was "rewritten + verified locally, awaiting force push".

What actually happened, from `git reflog`:

1. `filter-branch` rewrote `main` (`HEAD@{7}`), stripping the trailers.
2. Two commits landed on top of the rewrite (`0ac3f84`, `da4e0b0`).
3. A `git pull --rebase` (`HEAD@{4}`..`HEAD@{0}`) replayed those commits onto the
   **un-rewritten** `origin/main`, discarding the rewrite.

Current state:

| ref | commits | `Co-authored-by` trailers |
|---|---|---|
| `main` / `origin/main` (`e29bc66`) | 43 | **13** |
| `da4e0b0` (the rewrite, still in reflog) | 43 | **0** |
| `backup/pre-rewrite-31`, tag `pre-rewrite-31` (`9221e26`) | 41 | 13 |

`git diff da4e0b0 main` is empty — the rewritten branch has identical content and
only differs in commit metadata, so nothing has to be redone.

## To finish this ticket

Reflog entries expire (~90 days by default), so `da4e0b0` is not a permanent
artifact. Pin it first:

    git branch rewrite-31 da4e0b0

Then review and push:

    git log rewrite-31 --format='%h %ad  %s' --date=short   # review in full
    git log rewrite-31 --format='%(trailers:key=Co-authored-by)' | grep -c .   # expect 0
    git diff rewrite-31 main                                # expect empty
    git push --force-with-lease origin rewrite-31:main

`--force-with-lease` rather than `--force`: it refuses if `origin/main` moved since
the last fetch.

**Afterwards, set the branch to not re-run this problem:** `git pull --rebase`
against a rewritten remote is what undid it the first time. Once the push lands,
run `git fetch && git reset --hard origin/main` locally rather than pulling.

Still open in this ticket beyond the push: whether the five placeholder commit
subjects get reworded in the same pass.

**Comment by Dayallenr** [comment]
## Correction to the check above: `rewrite-31` is now one commit ahead of `main`

The earlier comment said `git diff rewrite-31 main` should come back empty. That
is no longer true, and a session that treats the mismatch as a problem will stall
on it.

`b758569` — "Retire the local working plan; the issue tracker is the plan" — was
committed onto `rewrite-31` rather than `main`, deliberately: committing it to
`main` would have meant the force push silently dropped it.

| ref | commits | `Co-authored-by` trailers |
|---|---|---|
| `main` / `origin/main` (`e29bc66`) | 43 | 13 |
| `rewrite-31` (`b758569`) | **44** | **0** |

`git diff main rewrite-31` is now exactly two files — `.gitignore` and
`docs/engineering-notes.md`. That is the expected difference, not a rewrite
artifact.

Revised pre-push checks:

    git log rewrite-31 --format='%(trailers:key=Co-authored-by)' | grep -c .   # expect 0
    git diff --stat main rewrite-31    # expect .gitignore + docs/engineering-notes.md only
    git push --force-with-lease origin rewrite-31:main

Everything else in the previous comment stands, including the note about not
using `git pull --rebase` afterwards.

**Comment by Dayallenr** [comment]
## State check, 2026-08-18 (later session) — the rewrite was lost a second time, and is now pinned to a tag

The previous comment said the rewrite was pinned to the local branch `rewrite-31`
(`b758569`, 44 commits, 0 trailers). That is no longer true. Verified against the refs:

    $ git reflog show rewrite-31
    e29bc66 rewrite-31@{0}: reset: moving to origin/main
    b758569 rewrite-31@{1}: commit: Retire the local working plan; the issue tracker is the plan
    da4e0b0 rewrite-31@{2}: branch: Created from da4e0b0

The branch was hard-reset onto `origin/main`, so `rewrite-31`, `main` and `origin/main`
all sit at `e29bc66` with **13** `Co-authored-by` trailers. This is the same class of
failure as the first loss (a `git pull --rebase` onto the un-rewritten remote): the
rewrite keeps being reachable only from a *branch reflog entry*, which expires.

**Nothing is lost.** `b758569` is still in the object store and still correct:

| ref | commits | `Co-authored-by` trailers |
|---|---|---|
| `main` / `origin/main` / branch `rewrite-31` (`e29bc66`) | 43 | 13 |
| `b758569` (the rewrite) | **44** | **0** |

`git diff --stat main b758569` is exactly the two expected files — `.gitignore` and
`docs/engineering-notes.md` — and `git diff main da4e0b0` is empty, so the rewritten
tree is still byte-identical to `main` apart from that one deliberate extra commit.

### It is now pinned to an annotated tag, not a branch

    git tag -a rewrite-31-final b758569

A tag does not move under `reset`, `pull --rebase`, or a checkout, which is what let this
get undone twice. Verified through the tag: 44 commits, 0 trailers.

### Revised pre-push checks (unchanged except for the ref name)

    git log rewrite-31-final --format='%h %ad  %s' --date=short          # review in full
    git log rewrite-31-final --format='%(trailers:key=Co-authored-by)' | grep -c .   # expect 0
    git diff --stat main rewrite-31-final    # expect .gitignore + docs/engineering-notes.md only
    git push --force-with-lease origin rewrite-31-final:main

Afterwards: `git fetch && git reset --hard origin/main`, **never** `git pull --rebase`.

Still open beyond the push: whether the five placeholder commit subjects get reworded in
the same pass.

**Comment by Dayallenr** [comment]
## The `rewrite-31-final` tag now carries one more commit — 45, still 0 trailers

`466747e` "Track the fine-tuning splits; they are not rebuildable off this machine"
was committed onto the rewrite lineage rather than onto `main`, for the same reason
`b758569` was: a commit on `main` is silently discarded by this ticket's force push.
The tag was moved to cover it.

| ref | commits | `Co-authored-by` trailers |
|---|---|---|
| `main` / `origin/main` (`e29bc66`) | 43 | 13 |
| `rewrite-31-final` (`466747e`) | **45** | **0** |

`git diff --stat main rewrite-31-final` is now seven files: the two from before
(`.gitignore`, `docs/engineering-notes.md`) plus `data/training/README.md`, the two
newly tracked splits, and the two sample files they replace.

Measured repository cost of the added data: **12.02 MiB → 15.14 MiB** packed.

Revised pre-push checks:

    git log rewrite-31-final --format='%h %ad  %s' --date=short          # review in full
    git log rewrite-31-final --format='%(trailers:key=Co-authored-by)' | grep -c .   # expect 0
    git diff --stat main rewrite-31-final    # expect the seven files above
    git push --force-with-lease origin rewrite-31-final:main

Afterwards: `git fetch && git reset --hard origin/main`, **never** `git pull --rebase`.

`ruff check` clean, 411 tests pass at the tag.

**Comment by Dayallenr** [comment]
## State check: the push has landed — `origin/main` is trailer-free

Verified today against the remote, not against this ticket's own history:

| ref | commits | `Co-authored-by` | `Generated with` |
|---|---|---|---|
| `origin/main` (`1fca287`) | 46 | **0** | **0** |
| `rewrite-31-final` (`2708cae`) | 45 | 0 | 0 |
| local `main` (`bede5e3`) | 47 | 0 | 0 |

Counted twice by different means — `git log --format='%(trailers:key=Co-authored-by)' | grep -c .`
and a raw `grep -ci` over full commit bodies (`%B`), which also catches a trailer that
lost its blank line and would not parse as one. Both return 0.

The tag is an ancestor of `origin/main`, which is ahead of it only by `1fca287`. So the
force push this ticket was gated on **has happened**, and the 13 trailers the earlier
comments record on `origin/main` are gone. This ticket's own state comments and
`CLAUDE.md` both still describe the pre-push world; correcting that here rather than
editing them, so the wrong version stays visible.

The seven `claude` matches remaining in commit bodies are references to `CLAUDE.md` and
`.claude/`, not tooling attribution — checked individually.

### What is still open

Only the sub-question the last comment left: **the five placeholder commit subjects**.
On `origin/main` today they are `training`, `update`, `answers`, `docs`, `queries`,
`update` — six, not five, since `1fca287` "training" landed after that comment. Rewording
them means another rewrite and another force push, so it is worth deciding whether it is
worth a second gated push at all, or whether the log is now good enough to leave alone.

### Not pushed

Local `main` is one commit ahead of `origin/main` (`bede5e3`, the #11 tracker fix). It is
a fast-forward, not a rewrite, and is left unpushed because nobody asked for it.

**Comment by Dayallenr** [comment]
## The seven placeholder subjects are reworded, and the push has landed

The open sub-question from the last comment is closed. `origin/main` is now `cff11ef`, 56 commits, **0 placeholder subjects**.

| old subject | new subject |
|---|---|
| `update` | Stop 10-Q tables of contents leaking into the table corpus |
| `queries` | Generate 4,776 synthetic training queries from the corpus |
| `docs` | Write the agent guides for the tracker and the domain docs |
| `answers` | Generate all 101 answers with the local model |
| `update` | Mine hard-negative triplets and add the training script |
| `training` | Record the fine-tuning run on the RTX 5070 |
| `update` | Commit the checkpoint digest manifest |

Each subject was written from that commit's own diff, not from memory. The `update` at `3abda1e` turned out to be the 10-Q table-of-contents fix — regex anchor plus per-row scoring, with the corpus regenerated — which is one of the more substantive commits in the history and was the least legible thing in the log.

**Bodies were left alone.** All seven had empty bodies; writing one now would be inventing history rather than restoring it. Subjects only.

### Verification, before the push

Rewrite was `git filter-branch --msg-filter` over `3abda1e^..HEAD`, keyed on **commit hash rather than subject text** — three of the seven are the literal string `update`, so matching on the text would have reworded whichever came past first.

- **Tree at `HEAD` byte-identical** — `git diff pre-reword-31 HEAD` is empty.
- **All 56 commits preserved**, none squashed.
- **Author and committer identity and both dates unchanged on every commit** — diffed `%an|%ae|%cn|%ce|%aI|%cI` across all 56, no differences.
- **Exactly seven subjects changed**, no others — diffed `%s` across all 56.
- `ruff check` clean, 563 tests pass after the rewrite.

### After the push, verified against the fetched remote

`origin/main` at `cff11ef`, local `HEAD` identical to it. Trailer audit re-run against the remote rather than the local copy: **0** `Co-authored-by` (counted both via `%(trailers:key=)` and a raw grep over full bodies, which also catches a trailer that lost its blank line), **0** `Generated with`, **0** one-word subjects, **0** placeholder subjects.

This push also published today's two commits from #25 (the model card and its code-review fixes), which were sitting unpushed.

### Recovery point

The pre-rewrite state is pinned as the annotated tag **`pre-reword-31`** (`d58b34c`), kept local and not pushed. A tag rather than a branch, per this ticket's own history — two branch pins were lost to `reset` and `pull --rebase`, and a tag does not move.

All nine of this ticket's user stories are now satisfied. Closing.
